esphome/components/esp32_touch/binary_sensor.py

```diff
@@ -15,23 +20,44 @@
 CONF_WAKEUP_THRESHOLD = "wakeup_threshold"
 
 TOUCH_PADS = {
-    4: cg.global_ns.TOUCH_PAD_NUM0,
-    0: cg.global_ns.TOUCH_PAD_NUM1,
-    2: cg.global_ns.TOUCH_PAD_NUM2,
-    15: cg.global_ns.TOUCH_PAD_NUM3,
-    13: cg.global_ns.TOUCH_PAD_NUM4,
-    12: cg.global_ns.TOUCH_PAD_NUM5,
-    14: cg.global_ns.TOUCH_PAD_NUM6,
-    27: cg.global_ns.TOUCH_PAD_NUM7,
-    33: cg.global_ns.TOUCH_PAD_NUM8,
-    32: cg.global_ns.TOUCH_PAD_NUM9,
+    VARIANT_ESP32: {
+        4: cg.global_ns.TOUCH_PAD_NUM0,
+        0: cg.global_ns.TOUCH_PAD_NUM1,
+        2: cg.global_ns.TOUCH_PAD_NUM2,
+        15: cg.global_ns.TOUCH_PAD_NUM3,
+        13: cg.global_ns.TOUCH_PAD_NUM4,
+        12: cg.global_ns.TOUCH_PAD_NUM5,
+        14: cg.global_ns.TOUCH_PAD_NUM6,
+        27: cg.global_ns.TOUCH_PAD_NUM7,
+        33: cg.global_ns.TOUCH_PAD_NUM8,
+        32: cg.global_ns.TOUCH_PAD_NUM9,
+    },
+    VARIANT_ESP32S2: {
+        1: cg.global_ns.TOUCH_PAD_NUM1,
+        2: cg.global_ns.TOUCH_PAD_NUM2,
+        3: cg.global_ns.TOUCH_PAD_NUM3,
+        4: cg.global_ns.TOUCH_PAD_NUM4,
+        5: cg.global_ns.TOUCH_PAD_NUM5,
+        6: cg.global_ns.TOUCH_PAD_NUM6,
+        7: cg.global_ns.TOUCH_PAD_NUM7,
+        8: cg.global_ns.TOUCH_PAD_NUM8,
+        9: cg.global_ns.TOUCH_PAD_NUM9,
+        10: cg.global_ns.TOUCH_PAD_NUM10,
+        11: cg.global_ns.TOUCH_PAD_NUM11,
+        12: cg.global_ns.TOUCH_PAD_NUM12,
+        13: cg.global_ns.TOUCH_PAD_NUM13,
+        14: cg.global_ns.TOUCH_PAD_NUM14,
+    },
 }
 
 
 def validate_touch_pad(value):
     value = gpio.validate_gpio_pin(value)
-    if value not in TOUCH_PADS:
-        raise cv.Invalid(f"Pin {value} does not support touch pads.")
+    variant = get_esp32_variant()
+    if variant not in TOUCH_PADS:
+        raise cv.Invalid(f"{variant} not supported")
+    if value not in TOUCH_PADS[variant]:
+        raise cv.Invalid(f"{variant} doesn't support ADC on pin {value}")
     return value
```

Comment:
I think if you change this, you don't need to look up the variant in to_code and can directly set my suggestion in my below comment. (Please do test this theory)
```suggestion
    return cv.enum(TOUCH_PADS[variant])(value)
```

Reply:
Thanks for the feedback! I got a bit sidetracked with other stuff but will fix this soon (if it's not already fixed in the main branch by now)